src/agent/src/config.rs

```diff
@@ -66,6 +72,8 @@ pub struct AgentConfig {
     pub unified_cgroup_hierarchy: bool,
     pub tracing: bool,
     pub supports_seccomp: bool,
+    pub https_proxy: String,
+    pub no_proxy: String,
```

Comment:
Just a comment, and probably we can think it further if we really get into some trouble.

Now we need `https_proxy` and `no_proxy` here because kata-agent handles image-pull requests. For kata-agent, this can be specified by kernel cmdline.

On TDX, kernel cmdline can be verified during remote attestation by RTMR.
However, on SNP, kernel cmdline is part of `measurement` of the evidence, this would make it harder for users who want to use a fixed rootfs + kernel + guest fw but dynamic `https_proxy`.

A potential way is to move the whole guest-pull logic (which involves `image-rs`) to another components and probably confidential data hub. Thus the function call routine would from
`kata-agent` -->`kata-agent's image-rs` -- image decryption --> `confidential-data-hub`'s image decryption API

to 

`kata-agent`-->`confidential-data-hub`'s image pull API --> `confidential-data-hub's image-rs`

The rationality is that 
1. `https_proxy` together with potential configs like [image_policy](https://github.com/kata-containers/kata-containers/blob/CCv0/src/agent/src/image_rpc.rs#L55), [image_registry_auth_file](https://github.com/kata-containers/kata-containers/blob/CCv0/src/agent/src/image_rpc.rs#L62C26-L62C50) could be move to CDH configuration file rather than kata-agent. Supposing that CDH's config file can be covered by initdata while kata-agent's cannot, these configs can be dynamically configured without changing the TEE's base measurement.
2. this can simplify the logic of kata-agent and make it a management and control component, while all data flow operations in the guest are handed over to CDH.

Reply:
Hey Ding, I think this makes a lot of sense. I just want to double check if you are proposing that we put a hold on this PR until #8870 that is the PR that starts up the CDH is merged and then the `kata-agent-->confidential-data-hub's image pull API` goes over a gRPC (or similar) API, or are you suggesting we pull in the CDH as a crate into kata-agent and call it via rust code?

Reply:
Steve, Ding has put forward a fantastic proposal that definitely deserves our full attention and a thorough discussion to achieve a shared agreement. Let’s aim to wrap up the code merge into main first, and then we can give this great idea the focus it needs.

Reply:
That's fine by me as long as we remember to tidy up any additions from this PR that would no longer be needed :)

Reply:
Yes. Let me make a separate issue about this, which should not block the current merge2main work and we can keep tracking that later.

Reply:
Done by https://github.com/kata-containers/kata-containers/issues/9266